src/readers/assembler/mod.rs

```diff
@@ -38,6 +38,21 @@ impl WebAPI {
         let parsed_toml: WebAPI = toml::from_str(toml_str).unwrap();
         parsed_toml
     }
+
+    /// Returns the list of endpoints (cleaner than having to actually access it directly)
+    ///
+    /// This iterates over all of the EndpointGroups and the pushes all of their endpoints onto
+    /// a `Vec<&Endpoint>`.
+    pub fn get_all_endpoints(&self) -> Vec<&Endpoint> {
+        let mut all_endpoints = Vec::new();
+        for group in &self.groups {
+            let group_endpoints = group.get_endpoints();
+            for endpoint in group_endpoints {
+                all_endpoints.push(endpoint);
+            }
+        }
+        all_endpoints
+    }
```

Comment:
This method returns ALL of the endpoints from the toml file, which is not optimal.

Since we want to separate the endpoints by group, this can't be a permanent fix. 

I just made it this way because we really just want to test the ability of the file writers, which are just based off of the `Endpoint` struct anyway, so swapping out to a better method later wont impact them at all (hopefully).

Anyway, writing with future refactor in mind.

Reply:
Can't you directly append group.get_end_points() to all_endpoints without the for loop